internal/provider/resource_edge_cluster.go

```diff
@@ -0,0 +1,245 @@
+				Description: "The nodes in the edge cluster",
+				Elem:        edge_cluster.EdgeNodeSchema(),
+			},
+		},
+	}
+}
+
+func resourceEdgeClusterCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	c := meta.(*api_client.SddcManagerClient).ApiClient
+
+	params := &nsxt_edge_clusters.CreateEdgeParams{
+		EdgeCreationSpec: edge_cluster.GetEdgeClusterCreationSpec(d),
+		Context:          ctx,
+	}
+
+	_, task, err := c.NSXTEdgeClusters.CreateEdge(params)
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	tflog.Info(ctx, "Edge cluster creation has started.")
+	err = meta.(*api_client.SddcManagerClient).WaitForTaskComplete(ctx, task.Payload.ID, false)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	var clusters *nsxt_edge_clusters.GetEdgeClustersOK
+	clusters, err = c.NSXTEdgeClusters.GetEdgeClusters(nil)
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	for _, cluster := range clusters.Payload.Elements {
+		if cluster.Name == d.Get("name") {
+			d.SetId(cluster.ID)
+			tflog.Info(ctx, "Edge cluster created successfully.")
+			return nil
+		}
+	}
+
+	return diag.Errorf("Edge cluster creation failed - cluster not found in inventory")
+}
+
+func resourceEdgeClusterRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	c := meta.(*api_client.SddcManagerClient).ApiClient
+
+	_, err := getEdgeCluster(ctx, c, d.Id())
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	return nil
+}
+
+func resourceEdgeClusterDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	tflog.Info(ctx, "Edge cluster deletion is not implemented. See KB article 78635 for more information.")
+	return nil
+}
+
+func resourceEdgeClusterUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	c := meta.(*api_client.SddcManagerClient).ApiClient
+
+	edgeClusterOk, err := getEdgeCluster(ctx, c, d.Id())
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	if d.HasChange("edge_node") {
+		oldNodesChange, newNodesChange := d.GetChange("edge_node")
+		oldNodesRaw, newNodesRaw := oldNodesChange.([]interface{}), newNodesChange.([]interface{})
+
+		if len(oldNodesRaw) == len(newNodesRaw) {
+			return diag.Errorf("Adding and removing edge nodes is not supported in a single configuration change. Apply each change separately.")
+		}
+
+		params := nsxt_edge_clusters.NewUpdateEdgeClusterParamsWithContext(ctx)
+		params.ID = d.Id()
+		params.EdgeClusterUpdateSpec = &models.EdgeClusterUpdateSpec{}
+
+		// Shrink
+		if len(oldNodesRaw) > len(newNodesRaw) {
+			operation := "SHRINKAGE"
+			params.EdgeClusterUpdateSpec.Operation = &operation
+			params.EdgeClusterUpdateSpec.EdgeClusterShrinkageSpec =
+				edge_cluster.GetEdgeClusterShrinkageSpec(edgeClusterOk.Payload.EdgeNodes, newNodesRaw)
+			tflog.Info(ctx, "Shrinking edge cluster")
+		}
+
+		// Expand
+		if len(oldNodesRaw) < len(newNodesRaw) {
+			operation := "EXPANSION"
```

Comment:
EXPANSION should be a constant

Reply:
Added a constant for EXPANSION. I had to keep the local "operation" variable because constants are not addressable and I can't use them directly in the spec